packages/abtest/src/error.ts

```diff
@@ -0,0 +1,46 @@
+/**
+ * Internal dependencies
+ */
+import { isDevelopmentMode } from './utils';
+
+export const logError = (
+	error: Record< string, string > & { message: string }
+): void => {
+	const onLoggingError = ( e: unknown ) => {
+		if ( isDevelopmentMode ) {
+			console.error( '[ExPlat] Unable to send error to server:', e ); // eslint-disable-line no-console
+		}
+	};
+
+	try {
+		const { message, ...properties } = error;
+		const logStashError = {
+			message,
+			properties: {
+				...properties,
+				context: 'explat',
+				explat_client: 'woocommerce',
+			},
+		};
+
+		if ( isDevelopmentMode ) {
+			console.error( '[ExPlat] ', error.message, error ); // eslint-disable-line no-console
+		} else {
+			// Bail if using IE 11.
+			if ( ! window.fetch ) {
+				return;
+			}
+
```

Comment:
Might want to add a tracks check in here as well as I have a feeling sending error data back to the server without tracking consent might not be ok:

```
if (!window.wcTracks.isEnabled) {
    throw new Error(`Tracking is disabled, can't send error to the server.`)
}
```